transcom/mymove
# Store distance on PPM update [delivers #157756011]

## Description

This PR modifies the server code to cache the computed mileage, planned and max allowed SIT, and incentive min and max on the PPM.

## Reviewer Notes

Steps to test this out:

```
$ make db_dev_reset
$ make db_dev_migrate
$ go run cmd/generate_test_data/main.go  -scenario 6
$ make server_run
```

Then, go through the Service member flow. You can use any values you want *except for the following*, which are required for the test data to work:

Origin zip: 94511
Destination zip: 78626
Weight: 7500-7550

After you complete the flow and sign the paperwork, check in your database that the values have been set:

```sql
SELECT mileage, planned_sit_max, sit_max, incentive_estimate_min, incentive_estimate_max from personally_procured_moves;
```

You should see something close to this:

```
mileage planned_sit_max	sit_max incentive_estimate_min incentive_estimate_max
1775    29400           189000  605204                 668910
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/157756011) for this change
